daskms/table_proxy.py

```diff
@@ -85,39 +101,64 @@ def proxied_method_factory(method, locktype):
     if locktype == NOLOCK:
         def _impl(table_future, args, kwargs):
             try:
-                return getattr(table_future.result(), method)(*args, **kwargs)
+                _impl.calls += 1
+                start_time = time()
+                result = getattr(table_future.result(), method)(*args, **kwargs)
+                end_time = time()
+                run_time.append(end_time - start_time)
+                return result
             except Exception:
                 if logging.DEBUG >= log.getEffectiveLevel():
                     log.exception("Exception in %s", method)
                 raise
+            finally:
+                _function_runs[method] = (_impl.run_time, _impl.calls)
```

Comment:
I think you can remove the finally clause, because you've modified the variables on the `_impl` object itself